pkg/session/txn.go

```diff
@@ -608,20 +608,56 @@ func (txn *LazyTxn) Wait(ctx context.Context, sctx sessionctx.Context) (kv.Trans
 		defer func(begin time.Time) {
 			sctx.GetSessionVars().DurationWaitTS = time.Since(begin)
 		}(time.Now())
+		if err := txn.waitWithSQLKiller(ctx, sctx); err != nil {
+			return txn, err
+		}
+		txn.lazyUniquenessCheckEnabled = !sctx.GetSessionVars().ConstraintCheckInPlacePessimistic
+	}
+	return txn, nil
+}
+
+func (txn *LazyTxn) waitWithSQLKiller(ctx context.Context, sctx sessionctx.Context) error {
+	finishCh := make(chan struct{})
+	defer close(finishCh)
 
+	tsoCh := make(chan error)
+	go func() {
```

Comment:
There will be some regression with the extra goroutine I think.

Test with sysbench random points:

```
10 threads
nightly: 8538 qps
this pr: 8308 qps

200 threads
nightly: 73670 qps
this pr: 71189 qps
```